File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientJava/src/PassiveEquipmentUpgradeModelView.java
import java.lang.*;

	public class PassiveEquipmentUpgradeModelView
	{

	
		public ArrayList<StateAlterationModelView> PassiveAlterations;	
		public float Cost;	
		public static PassiveEquipmentUpgradeModelView Deserialize(BufferedReader input) {
		try {
			PassiveEquipmentUpgradeModelView _obj =  new PassiveEquipmentUpgradeModelView();
			// PassiveAlterations
			ArrayList<StateAlterationModelView> _obj_PassiveAlterations = new ArrayList<StateAlterationModelView>();
			int _obj_PassiveAlterations_count = Integer.Parse(input.readLine());
			for(int _obj_PassiveAlterations_i = 0; _obj_PassiveAlterations_i < _obj_PassiveAlterations_count; _obj_PassiveAlterations_i++) {
				StateAlterationModelView _obj_PassiveAlterations_e = StateAlterationModelView.deserialize(input);
				_obj_PassiveAlterations.add((StateAlterationModelView)_obj_PassiveAlterations_e);
			}
			_obj.PassiveAlterations = _obj_PassiveAlterations;
			// Cost
			float _obj_Cost = Float.Parse(input.readLine());
			_obj.Cost = _obj_Cost;
			} catch (UnsupportedEncodingExceptio e) { 
			} catch (IOException e) { }
			return _obj;
		}

		public void serialize(OutputStreamWriter output) {
			try {
			// PassiveAlterations
			output.append(this.PassiveAlterations.size().toString() + "\n");
			for(int PassiveAlterations_it = 0; PassiveAlterations_it < this.PassiveAlterations.size();PassiveAlterations_it++) {
				this.PassiveAlterations[PassiveAlterations_it].serialize(output);
			}
			// Cost
			output.WriteLine(((Float)this.Cost).toString() + "\n");
			} catch (UnsupportedEncodingExceptio e) { 
			} catch (IOException e) { }
		}

	}
}

File: Codinsa2015.Client/Codinsa2015.Client/Views/PassiveEquipmentUpgradeModelView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework;


namespace Codinsa2015.Views
{

	public class PassiveEquipmentUpgradeModelView
	{

		static Encoding BOMLESS_UTF8 = new UTF8Encoding(false);
	
		// Obtient les altérations d'état appliquées passivement par cet équipement.
		public List<StateAlterationModelView> PassiveAlterations;	
		// Obtient le coût de l'upgrade.
		public float Cost;	
		public static PassiveEquipmentUpgradeModelView Deserialize(System.IO.StreamReader input) {
			PassiveEquipmentUpgradeModelView _obj =  new PassiveEquipmentUpgradeModelView();
			// PassiveAlterations
			List<StateAlterationModelView> _obj_PassiveAlterations = new List<StateAlterationModelView>();
			int _obj_PassiveAlterations_count = Int32.Parse(input.ReadLine());
			for(int _obj_PassiveAlterations_i = 0; _obj_PassiveAlterations_i < _obj_PassiveAlterations_count; _obj_PassiveAlterations_i++) {
				StateAlterationModelView _obj_PassiveAlterations_e = StateAlterationModelView.Deserialize(input);
				_obj_PassiveAlterations.Add((StateAlterationModelView)_obj_PassiveAlterations_e);
			}
			_obj.PassiveAlterations = (List<StateAlterationModelView>)_obj_PassiveAlterations;
			// Cost
			float _obj_Cost = Single.Parse(input.ReadLine());
			_obj.Cost = (float)_obj_Cost;
			return _obj;
		}

		public void Serialize(System.IO.StreamWriter output) {
			// PassiveAlterations
			output.WriteLine(this.PassiveAlterations.Count.ToString());
			for(int PassiveAlterations_it = 0; PassiveAlterations_it < this.PassiveAlterations.Count;PassiveAlterations_it++) {
				this.PassiveAlterations[PassiveAlterations_it].Serialize(output);
			}
			// Cost
			output.WriteLine(((float)this.Cost).ToString());
		}

	}
}
